rdt-app/src/main/java/io/ona/rdt_app/widget/RDTBarcodeFactory.java

```diff
@@ -95,20 +102,22 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
                                         for (String addr : rdtIdAddrs) {
                                             stepAndId = addr.isEmpty() ? new String[0] : addr.split(":");
                                             if (stepAndId.length == 2) {
-                                                jsonApi.writeValue(stepAndId[0].trim(), stepAndId[1].trim(), "RDT ID: " + barcodeValues[0].trim(), "", "", "", false);
+                                                jsonApi.writeValue(stepAndId[0].trim(), stepAndId[1].trim(), "RDT ID: " + barcodeValues[2].trim(), "", "", "", false);
                                             }
                                         }
 
                                         // populate exp. date to expiration date widget value
                                         stepAndId = expirationDateAddress.isEmpty() ? new String[0] : expirationDateAddress.split(":");
                                         if (stepAndId.length == 2) {
-                                            jsonApi.writeValue(stepAndId[0].trim(), stepAndId[1].trim(), barcodeValues[1].trim(), "", "", "", false);
+                                            try {
+                                                expDate = convertDate(barcodeValues[1].trim(), OPEN_RDT_DATE_FORMAT);
+                                                jsonApi.writeValue(stepAndId[0].trim(), stepAndId[1].trim(), getDateStr(expDate), "", "", "", false);
```

Comment:
Is the default Date format good enough. Why not use the ISO Date format for storing the date as a string

Reply:
The date is picked up as is from the rdt qr code.

It is currently in this format.

Reply:
Got that we get get the date from rdt using a custom format. But when saving on json we convert back to string using `date.toString();` using the method `getDateStr()`.

Reply:
Ok, done.